yt/frontends/amrvac/fields.py

```diff
@@ -29,14 +30,45 @@
     "spherical": ("r", "theta", "phi")
 }
 
+def _velocity(field, data, idir, prefix=None):
+    """Velocity = linear momentum / density"""
+    # This is meant to be used with functools.partial to produce
+    # functions with only 2 arguments (field, data)
+    # idir : int
+    #    the direction index (1, 2 or 3)
+    # prefix : str
+    #    used to generalize to dust fields
+    if prefix is None:
+        prefix = ""
+    moment = data["gas", "%smoment_%d" % (prefix, idir)]
+    rho = data["gas", "%sdensity" % prefix]
+
+    idx1 = np.where(rho == 0)[0]
```

Comment:
I think that here we could use the standard, rather than fancy, indexing.  Something like:

`idx1 = (rho==0)`

and then similarly for `idx2`.  You can get total with `.sum()` on the bool arrays, and you can then avoid doing the `in1d` by doing `np.all(idx1 == idx2)` by itself to check the True/False status.

Reply:
haha, it seems I can never find a legitimate use case for `np.where`. Every time I try to use it, it turns out there's a better way. Thanks !
Otherwise, I'm confused; why should I need the total of those boolean arrays ? then it's not as straight forward as the equality check you're suggesting because there can be 0s in `moment` where `density` is strictly positive (that would be a perfectly still patch of "fluid").

Reply:
Ah -- I was referring to the check of `idx1.size > 0`.  That also could be changed to `idx1.any()`.

Reply:
I like it, except I don't see how I can avoid using np.in1d + np.where in the following, though it's not a critical check that was mostly useful for me to get a sense of what I was doing. Now I guess I could simply remove it, would you agree ?

Reply:
What I *think* you can do is something like:

```
if idx1.any():
    mylog.info(...)
    idx2 = (moment==0)
    if not np.all(idx1 == idx2):
        raise RuntimeError
```

Unless I'm misunderstanding, and there's a possibility that `idx2` will have more 0 values than `idx1`.  If that's the case, you could do something like:

```
((idx1 & idx2) == idx1).all()
```

I think.

Reply:
Took me a few minutes to get my head around this but you are absolutely correct, this is exactly what's needed here. So now I should have covered all your suggestions :)